### ISSUE TITLE
about dataset

### ISSUE THREAD

**Issue by xiewende** [Haochen-Wang409/U2PL#47]
![Uploading image.png…]()

在这里是将有标签数据量扩充到无标签的数据量吗？如果是，这样扩充不会对网络训练产生影响吗？

**Comment by Haochen-Wang409** [comment]
您好，方便重新上传图片嘛？我这里好像无法显示

**Comment by xiewende** [comment]
在 u2pl/dataset/pascal_voc.py 中，下面代码里面，这样复制对半监督训练不会有影响嘛？

self.data_root = data_root
        self.transform = trs_form
        random.seed(seed)

        if len(self.list_sample) >= n_sup and split == "train":
            self.list_sample_new = random.sample(self.list_sample, n_sup)
        elif len(self.list_sample) < n_sup and split == "train":
            num_repeat = math.ceil(n_sup / len(self.list_sample))
            self.list_sample = self.list_sample * num_repeat
            self.list_sample_new = random.sample(self.list_sample, n_sup)
        else:
            self.list_sample_new = self.list_sample

**Comment by Haochen-Wang409** [comment]
这段代码本质不会进行复制，因为 ```len(self.list_sample)``` 永远会等于 ```n_sup```

**Comment by xiewende** [comment]
比如label=1464的例子
无标签的数据中 len(self.list_sample) 永远会等于 n_sup，但是有标签的是不行等的，有标签的数量是会扩充到无标签数量吗？
因为我看在主函数里得到的 train_loader_sup 和 train_loader_unsup的len是一样的，而且在训练的时候有标签的和无标签是会一起训练的，就是本来有标签的没那么多数据的，但是扩充到无标签数据的量的时候，会不会重复训练，这样对最终的效果是有影响的吗？还是我这边理解错误了？

**Comment by Haochen-Wang409** [comment]
有标签数据会扩充到与无标签数据一样的数量，这是一个半监督分割中常常采用的方法，以保证每个batch一半是有标签数据，一半是无标签数据。
例如 1464 有标签 + 9118 无标签，则有标签数据会过采样到 9118 张（即重复采样）

**Comment by xiewende** [comment]
好的，但是这样真的不会影响半监督训练吗？重复过多的有标签数量不会导致很依赖有标签的数据吗？

**Comment by Haochen-Wang409** [comment]
这个问题很有趣。仅凭借我个人的理解来看，对于有标签数据的过采样确实可能造成模型 biased to labeled data，但也保证了训练时的稳定性，减少了 unlabeled data 上错误伪标签的影响，这二者某种程度上是一个平衡。

为了缓解过拟合问题，一般会采用较强的数据扩增 (CutMix, ClassMix)，得到一张有标签图像 + 无标签图像的混合图像。

**Comment by xiewende** [comment]
理解了
谢谢
